api/server/agent.go

```diff
@@ -22,9 +23,12 @@ func (tapi ToDDApi) Agent(w http.ResponseWriter, r *http.Request) {
 	// Retrieve query values
 	values := r.URL.Query()
 
-	var agent_list []defs.AgentAdvert
-
-	var tdb = db.NewToddDB(tapi.cfg)
+	agent_list, err := tapi.tdb.GetAgents()
```

Comment:
It would probably help if I gave you an example of what that handler would look like if the code were correct: https://github.com/vCabbage/todd/blob/dc5427804243d9fb74981cf3401c110fa46d36b8/api/server/agent.go